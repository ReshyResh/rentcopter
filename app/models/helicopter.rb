class Helicopter < ApplicationRecord
  mount_uploader :avatar, AvatarUploader
  belongs_to :user
  has_many :reservations, dependent: :destroy

  validates :name, :description, :model, :image, presence: true
end
