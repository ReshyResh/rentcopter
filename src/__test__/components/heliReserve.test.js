import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import HeliReserve from '../../components/HeliReserve';
import helicopter from '../mockHeli';
import store from '../../redux/configureStore';

it('Helicopter modal component renders correctly', () => {
  const tree = renderer
    .create(<Router>
      <Provider store={store}>
        <HeliReserve helicopter={helicopter} />
      </Provider>
            </Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
