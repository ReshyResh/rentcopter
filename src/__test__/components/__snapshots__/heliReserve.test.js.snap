// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helicopter modal component renders correctly 1`] = `
<div
  className="h-full flex justify-center"
>
  <button
    className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root 
 !h-1/2
 !sm:h-1/2 
 !my-auto 
 !mr-3
 !rounded-full 
 !text-sm
 !bg-opacity-75
 !w-24
 !font-bold 
 !bg-yellow-600
 !hover:bg-indigo-700
 !transition-all
 !text-white css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    RESERVE
  </button>
</div>
`;
